python27-sys/build.rs

```diff
@@ -97,19 +93,143 @@ fn cfg_line_for_var(key: &str, val: &str) -> Option<String> {
     }
 }
 
+/// Run a python script using the 'python' located by PATH.
+fn run_python_script(script: &str) -> Result<String, String> {
+    let mut cmd = Command::new("python");
+    cmd.arg("-c").arg(script);
+
+    let out = try!(cmd.output().map_err(|e| {
+        format!("failed to run python interpreter `{:?}`: {}", cmd, e)
+    }));
+
+    if !out.status.success() {
+        let stderr = String::from_utf8(out.stderr).unwrap();
+        let mut msg = format!("python script failed with stderr:\n\n");
+        msg.push_str(&stderr);
+        return Err(msg);
+    }
+
+    let out = String::from_utf8(out.stdout).unwrap();
+    return Ok(out);
+}
+
+#[cfg(not(target_os="macos"))]
+fn get_rustc_link_lib(enable_shared: bool) -> Result<String, String> {
+    if enable_shared {
+        Ok("cargo:rustc-link-lib=python2.7".to_owned())
```

Comment:
your point about not hardcoding "python2.7" as the link target might apply here, though - I will see if we can extract this from sysconfig.